components/push/android/src/main/java/org/mozilla/push/PushConnection.kt

```diff
@@ -13,102 +13,84 @@ import java.io.File
 /**
  * An implementation of a [PushAPI] backed by a Rust Push library.
  *
- * @param path an absolute path to a file that will be used for the internal database.
+ * @param server_host the host name for the service (e.g. "push.service.mozilla.org").
+ * @param socket_protocol the optional socket protocol (default: "https")
+ * @param bridge_type the optional bridge protocol (default: "fcm")
+ * @param application_id the native OS messaging registration id
  * @param encryption_key an optional key used for encrypting/decrypting data stored in the internal
  *  database. If omitted, data will be stored in plaintext.
- //* /
-
-class PushConnection(path: String, encryption_key: String? = null) : PushAPI, AutoCloseable {
-    private var db: RawPushConnection?
+ */
+class PushConnection(
+    application_id: String,
+    sender_id: String,
+    server_host: String?="push.service.mozilla.org",
+    socket_protocol: String?="https",
+    bridge_type:String?="fcm",
```

Comment:
For the bridge_type, what are the valid types? Maybe it's easier with some typed class?

Reply:
I'll look into adding a typed class for this, but I'm definitely a Kotlin noob. I'll see if there's anything like an enumeration that could be used.